### PR TITLE
[chore] apply fieldalignment linter

### LINKED ISSUE #2789: Enable all disabled checks options in golangci-lint, see #2786
In #2786 we started to enable some of the disabled checks in golangci-lint, but not all of them are enabled because of the amount of work required.

This is a tracking issue to ensure all are enabled.

### PR DESCRIPTION
**Pull request by atoulme**
Fixes #2789

### CODE REVIEW

File: config/confighttp/confighttp.go [hunk 1]
```diff
@@ -245,28 +230,13 @@ func (interceptor *headerRoundTripper) RoundTrip(req *http.Request) (*http.Respo
...
-	// Additional headers attached to each HTTP response sent to the client.
-	// Header values are opaque since they may be sensitive.
-	ResponseHeaders map[string]configopaque.String `mapstructure:"response_headers"`
+	TLSSetting         *configtls.TLSServerSetting    `mapstructure:"tls"`
+	CORS               *CORSSettings                  `mapstructure:"cors"`
+	Auth               *configauth.Authentication     `mapstructure:"auth"`
+	ResponseHeaders    map[string]configopaque.String `mapstructure:"response_headers"`
+	Endpoint           string                         `mapstructure:"endpoint"`
+	MaxRequestBodySize int64                          `mapstructure:"max_request_body_size"`
+	IncludeMetadata    bool                           `mapstructure:"include_metadata"`
```
Comment (codeboten) [inline config/confighttp/confighttp.go:239]: where did all the docstrings go?

**Review by codeboten — COMMENTED** [review]
Pinging @mx-psi on this as he tried this originally and had some comments around the readability of the code w/ this linter enabled

File: config/confighttp/confighttp.go (hunk 1 above)
Comment (atoulme) [reply config/confighttp/confighttp.go:239]: I missed this one sorry, adding them back

Comment (atoulme) [reply config/confighttp/confighttp.go:239]: done

**Review by mx-psi — COMMENTED** [review]
We should at least exclude the test files, there's not much of a point there in terms of perf improvements.

IMO it does hinder readability in structs like the ones on confighttp and configgrpc, where I would expect that the Endpoint/Address comes first in the struct. It also does not seem to be very beneficial there. See also the other issues I pointed out on https://github.com/open-telemetry/opentelemetry-collector/issues/2789#issuecomment-974133991

Maybe we can start with enabling it just for `internal` packages? That seems like a good way to get started trying this out without worsening readability of the public API

**Comment by atoulme** [comment]
I can take slices of this PR and apply them selectively to the codebase to follow the pattern your recommend as separate PRs. It will make review easier.

**Comment by atoulme** [comment]
Closing as this won't land as is.

**Pull request closed without merging**